client/src/components/WaitingRoom/index.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase';
+import { compose } from 'recompose';
+import { withAuthorization } from '../Session';
+
+class WaitingRoomBase extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      players: [],
+      gameId: '',
+      timeLimit: 10,
+      started: false
+    };
+
+    this.onPlayerListUpdate = this.onPlayerListUpdate.bind(this);
+    this.onGameStart = this.onGameStart.bind(this);
+    this.startGame = this.startGame.bind(this);
+  }
+
+  async componentDidMount() {
+    //Grab gameId & timeLimit to set those as well as gameName
+    const { id } = this.props.match.params;
+    const gameInstance = await this.props.firebase.game(id).get();
+    this.setState({
+      gameId: id,
+      timeLimit: gameInstance.data().timeLimit,
+      started: gameInstance.data().hasStarted,
+      players: gameInstance.data().players
+    });
+  }
+
+
+
+  //Listens for an update to players
+  async onPlayerListUpdate(gameId) {
+    const game = this.props.firebase.game(gameId).get();
+    game.on('child_added', async function(snapshot)
+    {
+      this.setState({
+        players: snapshot.data().players
+      })
+    });
+  }
+
+  async onGameStart(gameId) {
+    const game = await this.props.firebase.game(gameId).get();
+
+    console.log('started');
+    game.on('value', (snapshot) => {
+      this.setState({
+        started: snapshot.data().hasStarted
+      })
+    })
+  }
+
+  async startGame(gameId){
+    const game = this.props.firebase.db.doc(`games/${gameId}`);
+    game.update({
+      hasStarted: true
+    });
+  }
+
+
+
+  render() {
+    const { gameId, players, timeLimit, started } = this.state
+    console.log(players)
+    return(
+      <div>
+        <h3>Time Limit Per Turn: {timeLimit}</h3>
+        <nav>
+          <header>Host: {players[0]}</header>
+          <header>Players:</header>
+          <ul style={{'listStyle': 'none'}}>
+            {
+              players.slice(1,players.length)
+                .map((player) => <li>{player}</li>)
+            }
+          </ul>
+        </nav>
+        <button type="button" onClick={() =>
+          this.startGame(gameId)}>
+          {started ? 'Join' : 'Start'}
```

Comment:
I'm a little confused by this logic. Does this mean anyone can start the game?

Reply:
You will need access to the `authUser` object to perform the check to determine whether the current logged in user is the host of the game.

Reply:
at this point, that was the case. has since been fixed

Reply:
Could you point me to the line(s) where you are doing this check? It's a lot of code so I may have missed it while skimming